models/badges.js

```diff
@@ -27,6 +27,35 @@ const fetchBadges = async ({
   }
 }
 
+/**
+ * Fetches the data about user badges
+ * @param query { Object }: Filter for badges data
+ * @return {Promise<userBadgeModel|Array>}
+ */
+
+const fetchUserBadges = async (username) => {
+  try {
+    const snapshot = await badgeModel.get()
+    const allBadges = []
+    snapshot.forEach((doc) => {
+      allBadges.push(doc.data())
+    })
+    const userBadges = []
+    allBadges.forEach((badge) => {
+      badge.users.forEach((user) => {
+        if (user === username) {
+          userBadges.push({ title: badge.title, description: badge.description })
```

Comment:
So much of stuff can be avoided
You can use `array-contains` or similar query provided by firestore 😉

Reply:
Ahh yes, I'll make the changes.